### PR TITLE
Update docs for contacts.yaml

### PR DESCRIPTION
**Pull request by zhijie-yang**
- [x] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] I am following the [CONTRIBUTING](https://github.com/canonical/oci-factory/blob/main/CONTRIBUTING.md) guidelines

*Ping the @canonical/rocks team.*

---

## Description
This PR adds the descriptions of the `maintainers` field in the `contacts.yaml` file and corresponding onboarding instructions for adding the maintainers.

---

*Picture of a cool rock:*

### CODE REVIEW

File: tools/cli-client/README.md [hunk 1]
```diff
@@ -10,13 +10,13 @@ A CLI client that triggers GitHub workflows for building, uploading and releasin
 
 See ["How to Contribute as a Maintainer"](https://github.com/canonical/oci-factory?tab=readme-ov-file#as-a-maintainer--).
 
-Upon finishing the onboarding, the user should receive a GitHub Personal Access Token. This token grants you proper
-permissions to trigger the build, upload and release of your rocks. This token should never be shared with 
-third-parties, nor put anywhere that is publicly available.
+Upon finishing the onboarding, the user should add their GitHub username to the `rockcraft.yaml` file in the
+`maintainers` field of the corresponding rock. This field is a list of GitHub usernames that are allowed to
```
Comment (clay-lake) [inline tools/cli-client/README.md:14]: Should we add a reference to the README.md with the sample contacts.yaml?

File: tools/cli-client/README.md [hunk 2]
```diff
@@ -10,13 +10,13 @@ A CLI client that triggers GitHub workflows for building, uploading and releasin
 
 See ["How to Contribute as a Maintainer"](https://github.com/canonical/oci-factory?tab=readme-ov-file#as-a-maintainer--).
 
-Upon finishing the onboarding, the user should receive a GitHub Personal Access Token. This token grants you proper
-permissions to trigger the build, upload and release of your rocks. This token should never be shared with 
-third-parties, nor put anywhere that is publicly available.
+Upon finishing the onboarding, the user should add their GitHub username to the `rockcraft.yaml` file in the
```
Comment (clay-lake) [inline tools/cli-client/README.md:13]:
```suggestion
Upon finishing the onboarding, the user should add their GitHub username to the `contacts.yaml` file in the
```

**Review by clay-lake — CHANGES_REQUESTED** [review]

File: tools/cli-client/README.md (hunk 1 above)
Comment (zhijie-yang) [reply tools/cli-client/README.md]: Thx, added.

**Review by clay-lake — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -322,6 +322,7 @@ The `contacts.yaml` file has the following schema:
 | notify | False | Dict[str, List[str]] | Who and how to notify about the build progress. |
 | notify.emails | False | List[str] | List of emails that should receive notifications. |
 | notify.mattermost-channels | False | List[str] | Mattermost channel **IDs**. |
+| maintainers | False | List[str] | List of maintainers' GitHub usernames, who are permitted to trigger workflows against this image. |
```
Comment (cjdcordeiro) [inline README.md:325]:
```suggestion
| maintainers | False | List[str] | The maintainers' GitHub usernames who can trigger workflows against this image. |
```

File: README.md [hunk 4]
```diff
@@ -332,6 +333,9 @@ notify:
 
   mattermost-channels:
     - fbdezwkcxpfofpysjore1wpfoc
+
+maintainers:
+  - foo
```
Comment (cjdcordeiro) [inline README.md:338]:
```suggestion
  - octocat
```

Just to be consistent with the GH user docs :p

**Review by cjdcordeiro — APPROVED** [review]
LGTM, just left a couple of possible improvements ;)

**Pull request merged**